src/lib/mdns/MdnsCache.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <cstdint>
+
+#include <core/CHIPError.h>
+#include <core/PeerId.h>
+#include <inet/IPAddress.h>
+#include <inet/InetInterface.h>
+#include <inet/InetLayer.h>
+#include <system/SystemTimer.h>
+#include <system/TimeSource.h>
+
+// set MDNS_LOGGING to enable logging -- sometimes used in debug/test programs -- traces the behavior
+#ifdef MDNS_LOGGING
+#define MdnsLogProgress ChipLogProgress
+#else
+#define MdnsLogProgress(...)
+#endif
+
+namespace chip {
+namespace Mdns {
+
+template <size_t CACHE_SIZE>
+class MdnsCache
+{
+public:
+    MdnsCache() : elementsUsed(CACHE_SIZE)
+    {
+        nullPeerId.SetFabricId(kUndefinedFabricId);
+        nullPeerId.SetNodeId(kUndefinedNodeId);
+
+        for (MdnsCacheEntry & e : mLookupTable)
+        {
+            // each unused entry decrements the count
+            MarkEntryUnused(e);
+        }
+        MdnsLogProgress(Discovery, "construct mdns cache of size %ld", CACHE_SIZE);
+    }
+
+    // insert this entry into the cache.
+    // return error if cache is full
+    // TODO:   have an eviction policy so if the cache is full, an entry may be deleted.
+    //         One policy may be Least-time-to-live
```

Comment:
Do we need to leave this as a TODO?  I would think the obvious policy would simply be to evict the oldest entry.

This doesn't seem like something we need to wait to add.

Reply:
I'm not sure least-time-to-live is what we want, since we need a way to mark "interesting" nodes, and ignore "non-interesting nodes" (not all nodes would be treated the same way).

Reply:
Sure, that's fair.